python/lsst/resources/tests.py

```diff
@@ -28,6 +28,19 @@
 TESTDIR = os.path.abspath(os.path.dirname(__file__))
 
 
+def prepare_testing(testcase: unittest.TestCase) -> None:
+    endpoint = os.environ.get("S3_ENDPOINT_URL")
+
+    if not endpoint:
+        return
+    os.environ["S3_ENDPOINT_URL"] = ""
+
+    def cleanup() -> None:
+        os.environ["S3_ENDPOINT_URL"] = Optional[endpoint]
```

Comment:
This looks wrong. `Optional[]` is a type annotation and is not something that should be assigned.

Mypy is complaining that the variable `endpoint` is `str | None` and so telling you that you can't assign `None` to `os.environ` which is defined as `dict[str, str]`. You can fix this by changing the code to be:

```python
if endpoint:
    os.environ["S3_ENDPOINT_URL"] = endpoint
```

Reply:
That should say `if endpoint is not None` or course.